bicytok/distanceMetricFuncs.py

```diff
@@ -1,641 +1,221 @@
-                + weightDF.loc[weightDF["Receptor"] == "IL2Ra", "Weight"].values[0]
-                + weightDF.loc[weightDF["Receptor"] == "IL2Rb", "Weight"].values[0]
-            ) / 3
-            conversion_factor = avg_weight
-        target_receptor_counts[:, 0] *= conversion_factor_sig
-        off_target_receptor_counts[:, 0] *= conversion_factor_sig
-
-        target_receptor_counts[:, 1] *= conversion_factor
-        off_target_receptor_counts[:, 1] *= conversion_factor
-
-        average_receptor_counts = np.mean(
-            np.concatenate((target_receptor_counts, off_target_receptor_counts)), axis=0
-        )
-        # Normalize the counts by dividing by the average
-        if (
-            average_receptor_counts[0] > 5
-            and average_receptor_counts[1] > 5
-            and np.mean(target_receptor_counts[:, 0])
-            > np.mean(off_target_receptor_counts[:, 0])
-            and np.mean(target_receptor_counts[:, 1])
-            > np.mean(off_target_receptor_counts[:, 1])
-        ):
-            target_receptor_counts = (
-                target_receptor_counts.astype(float) / average_receptor_counts
-            )
-            off_target_receptor_counts = (
-                off_target_receptor_counts.astype(float) / average_receptor_counts
-            )
-            # Matrix for emd parameter
-            M = ot.dist(target_receptor_counts, off_target_receptor_counts)
-            # optimal transport distance
-            a = (
-                np.ones((target_receptor_counts.shape[0],))
-                / target_receptor_counts.shape[0]
-            )
-            b = (
-                np.ones((off_target_receptor_counts.shape[0],))
-                / off_target_receptor_counts.shape[0]
-            )
-            optimal_transport = ot.emd2(a, b, M, numItermax=1000000000)
-            if special_receptor is not None:
-                return optimal_transport  # Return the distance value directly
-            results.append((optimal_transport, receptor_name, signal_receptor))
-        else:
-            results.append((0, receptor_name, signal_receptor))
-
-    # end loop
-    sorted_results = sorted(results, reverse=True)
-    top_receptor_info = [
-        (receptor_name, optimal_transport, signal_receptor)
-        for optimal_transport, receptor_name, signal_receptor in sorted_results[:10]
-    ]
-    # bar graph
-    receptor_names = [info[0] for info in top_receptor_info]
-    distances = [info[1] for info in top_receptor_info]
-    if ax is not None:
-        ax.bar(range(len(receptor_names)), distances)
-        ax.set_xlabel("Receptor")
-        ax.set_ylabel("Distance")
-        ax.set_title("Top 5 Receptor Distances (2D)")
-        ax.set_xticks(range(len(receptor_names)))
-        ax.set_xticklabels(receptor_names, rotation="vertical")
-
-    return sorted_results
-
-
-def EMD_3D(dataset1, target_cells, ax=None):
+
+    KL_div_vals = np.full(recAbundances.shape[1], np.nan)
```

Comment:
I see that you're filling in values as nan elsewhere. You can just initialize to nan like this.

Reply:
Done